src/System.ComponentModel.TypeConverter/src/System/ComponentModel/AddingNewEventArgs.cs

```diff
@@ -0,0 +1,59 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+
+
+
+
+namespace System.ComponentModel
+{
+    using System;
+    using System.ComponentModel;
+    using System.Security.Permissions;
+
+    /// <summary>
+    ///     Provides data for an event that signals the adding of a new object
+    ///     to a list, allowing any event handler to supply the new object. If
+    ///     no event handler supplies a new object to use, the list should create
+    ///     one itself.
+    /// </summary>
+    [HostProtection(SharedState = true)]
+    public class AddingNewEventArgs : EventArgs
```

Comment:
I am going to capture these and the related comments about improving the code into a separate issue that we can tackle after the code has been ported into the open.